full-service/src/json_rpc/v2/api/wallet.rs

```diff
@@ -446,15 +460,20 @@ where
                 None => None,
             };
 
-            let unsigned_tx_proposal: UnsignedTxProposal = (&service
-                .build_transaction(
+            let unsigned_tx_proposal: UnsignedTxProposalJSON = (&service
+                .build_unsigned_transaction(
                     &account_id,
                     &addresses_and_amounts,
                     input_txo_ids.as_ref(),
                     fee_value,
                     fee_token_id,
                     tombstone_block,
                     max_spendable_value,
+                    // TODO - We can check if the account has an account key to sign with, and if
+                    // so, we can include an RTH memo.
+                    // For hardware-wallet or view-only accounts, we can't include an RTH memo.
+                    // For now the behavior is to just not include a memo when doing offline
+                    // signing.
```

Comment:
The existing behavior (before this PR) is to not include memos in unsigned transactions, so I kept it. There are a few options here:
1. Leave as-is
2. Include RTH memos only if we're building an unsigned transaction for a non-view-only account
3. Include RTH memos in either case, which will result in the destination memo being generated but not the authenticated sender memo for view-only-accounts.
I don't have a strong feeling one way or the other since I'm not sure how the `build_unsigned_transaction` API is used in the wild.